deploy.py

```diff
@@ -56,34 +78,88 @@ def allow_tcp(port, from_ip="0.0.0.0/0"):
         self._template = Template()
         self._template.add_version("2010-09-09")
+        self.use_firehose = use_firehose or use_processor
+        self.use_processor = use_processor
         self.add_resource = self._template.add_resource
         self.add_parameter = self._template.add_parameter
 
         self.AutopushVersion = self.add_parameter(Parameter(
             "AutopushVersion",
             Type="String",
             Description="Autopush version to deploy",
-            Default="1.14.1",
+            Default="1.14.2",
             AllowedValues=[
+                "latest",
+                "1.14.2",
                 "1.14.1",
             ]
         ))
-        self.PushVPC = self.add_parameter(Parameter(
-            "ExistingVPC",
-            Type="AWS::EC2::VPC::Id",
-            Description=(
-                "The VPC Id so launch the Autopush server into"
+
+        if self.use_processor:
+            self.ProcessorLambdaBucket = self.add_parameter(Parameter(
+                "ProcessorLambdaBucket",
+                Type="String",
+                Default="push-lambda-funcs",
+                Description="S3 Bucket of lambda Message Processor",
+            ))
+            self.ProcessorLambdaKey = self.add_parameter(Parameter(
+                "ProcessorLambdaKey",
+                Type="String",
+                Default="push_processor_0.4.zip",
+                Description="S3 Key of lambda Message Processor",
+            ))
+            self.ProcessorVPCId = self.add_parameter(Parameter(
+                "ProcessorVPCId",
+                Type="AWS::EC2::VPC::Id",
+                Description="VPC to run the processor/Messages API in"
+            ))
+            self.ProcessorSubnetId = self.add_parameter(Parameter(
+                "ProcessorSubnetIds",
+                Type="List<AWS::EC2::Subnet::Id>",
+                Description=(
+                    "Subnet(s) to run Lambda processor in, MUST have "
+                    "a default route to a NAT Gateway"
                 ),
-        ))
+            ))
+            self.MessagesSecurityGroup = self.add_parameter(Parameter(
+                "MessagesSecurityGroup",
+                Type="AWS::EC2::SecurityGroup::Id",
+                Description="Security Group to allow Messages API access from",
+            ))
+            self.MessageAPISubnetId = self.add_parameter(Parameter(
+                "MessageApiEC2Subnet",
+                Type="AWS::EC2::Subnet::Id",
+                Description=(
+                    "Subnet to run Push Messages EC2 Instance in, MUST "
+                    "be in the same VPC as the Processor Subnets"
+                )
+            ))
+            self.PushMessagesVersion = self.add_parameter(Parameter(
+                "PushMessagesVersion",
+                Type="String",
+                Description="Push-Messages API version to deploy",
+                Default="0.5",
+                AllowedValues=[
```

Comment:
comment: blech, seems like this could grow pretty unwieldy fast. 
Sadly, not sure you can do much to fix that.